shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -99,10 +136,10 @@ FLUTTER_EXPORT
  * immediately.
  *
  * @param entrypoint The name of a top-level function from the same Dart
- *   library that contains the app's main() function.  If this is nil, it will
- *   default to `main()`.  If it is not the app's main() function, that function
- *   must be decorated with `@pragma(vm:entry-point)` to ensure the method is not
- *   tree-shaken by the Dart compiler.
+ *   library that contains the app's main() function.  If this is FlutterDefaultDartEntrypoint (or
+ *   nil) it will default to `main()`.  If it is not the app's main() function, that function must
```

Comment:
Are you sure you want to allow `nil` here? I explicitly didn't allow that on the Android side. Instead I offered a factory method for `DartEntrypoint.createDefault()` to make it easy to resolve, but still explicitly selecting the entrypoint.

Reply:
I don't like the interface and I don't like the use of nil as placeholders for future calculations either.  In this PR I've tried to make it so no one ever has to mention a nil to create or run an Engine.  Truly eliminating the use of nil will require some excavation, a breaking change and some philosophy debates.  I've avoided that to make some concrete progress on the issue.

Reply:
Ok.